### ISSUE TITLE
[Tracking] Docs rebuild

### ISSUE THREAD

**Issue by broccolini** [primer/doctocat#107]
## What and why
The current style guide was built to be home to multiple _guides_ yet only Primer CSS documentation is regularly maintained. The current site is built with Jekyll and we stretched it to accommodate multiple sites which has made it difficult to maintain and created a high barrier to contributors. Having the documentation site living in a different repo to Primer CSS code creates some difficulties and makes it inefficient to update and contributes to missing documentation and bugs. Concurrently we re-branded Primer last year to primer.style and have been moving all sites under this domain, so we should do that for Primer CSS to be consistent.

This project will focus on moving the Primer CSS documentation website from https://github.com/github/styleguide/ to http://github.com/primer/primer as a precursor to deprecating the style guide.

## Challenges / known issues
* We are likely to build similar components to those used on Components and other Primer websites which won't feel very DRY, though it's too early to abstract patterns. However, this project will help inform future projects to build a boilerplate for Primer websites.
* We need to re-think the Primer CSS homepage content and design to work better with the new primer.style website.
* We're still learning how best to use technologies like Next.js.
* We only made the style guide public a year ago, we need to make sure the new site is easy to find.
* We probably have a lot of links to update in other GitHub documentation.
* We don't have many people we can put on this project full time so Shawn will require support via feedback and code review.

## Who
* @shawnbot will lead the development work.
* @emplums will provide code review and feedback on docs site build.
* @broccolini will provide guidance on scope, design, and homepage content.
* @emilybrick will provide feedback on design and homepage content.

## Scope
Tasks are prioritized into Must, Should, Could, and Won't.

### Must

These tasks are considered required in order for the project to be successful. They must be completed before publicly shipping the updated documentation site publicly.

* [x] All documentation ported over (after primer/css#640 merges)
    * [x] Docs
    * [x] Getting Started
    * [x] Principles
    * [x] Tools
    * [x] Remove “What's New” and add nav link called “Releases” and point to [releases](https://primer-css-iizfqkcauw.now.sh/css/whats-new/changelog) directly on GitHub. 
    * [x] Move “Status key” to “Docs” as level 2 item above “Support”
* [x] All documentation links such as status, npm packages, and view source should work
* [x] Code examples should work with correct syntax highlighting
* [x] Implement search (same or better functionality as current style guide)
* [x] Responsive navigation, showing top level items (enhancement upon current style guide_)
* [ ] Design
    * [x] Custom illustration for banner
    * [ ] Updated homepage styles [tbd @broccolini]
    * [x] Top level nav matches primer.style/components
    * [x] Secondary (left) nav matches current styleguide, but build it better with primer components, and remove status icons
* [ ] Updated homepage content [tbd @broccolini]
* [ ] Re-direct styleguide/primer to new primer.style url

### Should

These tasks should be completed for the project to be seen as “done”, but do not need to be completed before making the new site public. These tasks can be addressed in follow-up sprints if not included in the initial ship.

* [ ] Update SCSS principles
    * [ ] Remove “Guidelines for using Sass features”
    * [x] Update “Lint rules” and link to [Tools > Linting](https://primer-css-iizfqkcauw.now.sh/css/tools/linting)
* [ ] Update contributing Guidelines to match new workflow
* [x] Update repo to github.com/primer/css to match new site url
* [x] Update npm to point to @primer/css to match new site url
* [ ] Add notice to style guide homepage that is being deprecated
* [x] Update references of “Primer” to be “Primer CSS” (where it makes sense to)
* [ ] Design global solution for feedback that works on similar primer documentation sites
* [ ] Deprecate http://styleguide.github.com [tbd @broccolini]
* [x] Document/capture which components should be abstracted to use across other Primer websites (such as navigation)
* [ ] As far as possible, ensure inbound links are updated to the new url

### Could

These tasks are “nice to have's” but not required for project completion. They can be completed adhoc in future Primer releases, though they may be elevated in priority depending on feedback.

* [x] Add live code editor like Primer Components
* [ ] Update Principles to be more relevant code-style principles
* [ ] Add intro to Tools index
* [ ] Add cross-links to primer.style website
* [ ] Review and update component status

### Won't

These items will not be part of the project scope but may be part of future Primer CSS releases.

* Move new components into Primer that live in GitHub.com
* Remove Accessibility docs (until Design website is live)
* Build reusable components to use across similar Primer websites.
* Design a solution for navigating between different Primer websites.
